dms/models/dms_file.py

```diff
@@ -638,7 +638,12 @@ def _inverse_content(self):
 
     def _create_model_attachment(self, vals):
         res_vals = vals.copy()
-        directory = self.env["dms.directory"].sudo().browse(res_vals["directory_id"])
+        directory = False
+        directory_model = self.env["dms.directory"]
+        if "directory_id" in res_vals:
+            directory = directory_model.browse(res_vals["directory_id"])
+        elif self.env.context.get("active_id"):
+            directory = directory_model.browse(self.env.context.get("active_id"))
```

Comment:
Is more readable?
<pre>
directory_id = res_vals.get("directory_id", self.env.context.get("active_id"))
directory = directory_id and self.env["dms.directory"].browse(directory_id)
</pre>